### PR TITLE
Add OS distro info to ProcessInfo event

### PR DESCRIPTION
**Pull request by sywhang**
For https://github.com/dotnet/runtime/issues/37374.

This adds the OS distro info to EventPipeEventSource's ProcessInfo Event.

### CODE REVIEW

File: src/coreclr/src/vm/eventpipeeventsource.cpp [hunk 1]
```diff
@@ -30,10 +40,12 @@ EventPipeEventSource::EventPipeEventSource()
...
-    const unsigned int numParams = 1;
+    const unsigned int numParams = 2;
     EventPipeParameterDesc params[numParams];
     params[0].Type = EventPipeParameterType::String;
     params[0].Name = W("CommandLine");
+    params[1].Type = EventPipeParameterType::String;
+    params[1].Name = W("OSInformation");
 
     size_t metadataLength = 0;
     BYTE *pMetadata = EventPipeMetadataGenerator::GenerateEventMetadata(
```
Comment (noahfalk) [inline src/coreclr/src/vm/eventpipeeventsource.cpp:51]: I recommend bumping the version number from 0 to 1

**Review by noahfalk — APPROVED** [review]
LGTM as long as you bump the version

**Review by josalem — APPROVED** [review]
Looks good to me as well!  Do we already send arch information in another event?

File: src/coreclr/src/vm/eventpipeeventsource.cpp [hunk 2]
```diff
@@ -17,6 +17,16 @@
 const WCHAR* EventPipeEventSource::s_pProviderName = W("Microsoft-DotNETCore-EventPipe");
 const WCHAR* EventPipeEventSource::s_pProcessInfoEventName = W("ProcessInfo");
 
+#if defined(HOST_WINDOWS)
+const WCHAR* EventPipeEventSource::s_pOSInformation = W("Windows");
+#elif defined(__APPLE__)
+const WCHAR* EventPipeEventSource::s_pOSInformation = W("OSX");
```
Comment (danmosemsft) [inline src/coreclr/src/vm/eventpipeeventsource.cpp:23]: `macOS` ?

**Pull request merged**